src/installer/pkg/sfx/installers/host.wxs

```diff
@@ -47,6 +62,16 @@
     <DirectoryRef Id="$(var.Program_Files)">
       <Directory Id="PROGRAMFILES_DOTNET" Name="dotnet" />
     </DirectoryRef>
+    
+    <?if $(var.Platform)~=x64 ?>
+    <CustomActionRef Id="Set_PROGRAMFILES_DOTNET_NON_NATIVE_ARCHITECTURE" />
+    <?endif?>
   </Fragment>
-
-</Wix>
+  
+  <Fragment>
+    <!-- Unlike DOTNETHOME which gives precedence to a user specified value over an x64 suffix, here we always want the suffixed path -->
+    <SetProperty Action="Set_PROGRAMFILES_DOTNET_NON_NATIVE_ARCHITECTURE" Id="PROGRAMFILES_DOTNET" Value="[$(var.Program_Files)]dotnet\x64\" After="Set_NON_NATIVE_ARCHITECTURE">
```

Comment:
Is this action defined in a separate file?

Reply:
Yes https://github.com/dotnet/arcade/pull/7785/files#diff-0938a95b0962aaae248064ee37481237c283dac60ea91ebc3ef38138bbbb52fbR23